src/main/java/com/mercuryCyclists/Sales/service/OnlineSaleService.java

```diff
@@ -119,7 +145,18 @@ public OnlineSale registerBackorder(OnlineSale onlineSale){
         }
 
         onlineSaleRepository.save(onlineSale);
-        kafkaTemplate.send("sales", new Gson().toJson(onlineSale));
+
+        // Create and populate sale event
+        SaleEvent saleEvent = new SaleEvent();
+        saleEvent.setProductName(product.get("name").getAsString());
+        saleEvent.setQuantity(onlineSale.getQuantity());
+        Double pricePerProduct = product.get("price").getAsDouble();
+        Double totalSalePrice = pricePerProduct * onlineSale.getQuantity();
+        saleEvent.setPrice(totalSalePrice);
+
+        System.out.println(saleEvent);
```

Comment:
can we make this a generic method in sale service, then call it from all.
rm println also